### ISSUE TITLE
[Bug]: DATA RACE during bootstrap

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#9895]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):b1cafe
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

logs can be found here:https://github.com/matrixorigin/matrixone/actions/runs/5194925610
```
2023/06/07 01:20:36.051804 +0000 INFO dn-service.logtail-push-service service/session.go:412 send subscription response {"uuid": "dd4dccb4-4d3c-41f8-b482-5251dc7a41bf", "server-id": "09bde5be-a69a-468f-b602-a92ec23b5ebd", "stream-id": 1, "remote": "127.0.0.1:55228", "table": "db_id:1 tb_id:1 ", "To": "PhysicalTime:1686100836040756747 "}
==================
WARNING: DATA RACE
Read at 0x00c0000bd920 by goroutine 40:
  runtime.mapiterinit()
      /opt/hostedtoolcache/go/1.20.4/x64/src/runtime/map.go:815 +0x0
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.(*Registry).ExportLog()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:48 +0xa4
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).gatherAndWrite()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:85 +0x64
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:60 +0x164

Previous write at 0x00c0000bd920 by main goroutine:
  runtime.mapassign_faststr()
      /opt/hostedtoolcache/go/1.20.4/x64/src/runtime/map_faststr.go:203 +0x0
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.(*Registry).Register()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:35 +0x1c5
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.Register()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:79 +0x213
  main.(*Config).createFileService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/config.go:213 +0x1cc
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:141 +0x11d
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:127 +0x3b3
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:52 +0x244
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:90 +0x367

Goroutine 40 (running) created at:
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:53 +0x16a
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.InitMetric()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/metric.go:101 +0xd04
  main.initTraceMetric.func2()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:377 +0x155
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:259 +0xd1
==================
==================
WARNING: DATA RACE
Read at 0x00c005830990 by goroutine 40:
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.(*Registry).ExportLog()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:48 +0xef
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).gatherAndWrite()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:85 +0x64
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:60 +0x164

Previous write at 0x00c005830990 by main goroutine:
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.(*Registry).Register()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:35 +0x1d1
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.Register()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:79 +0x213
  main.(*Config).createFileService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/config.go:213 +0x1cc
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:141 +0x11d
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:127 +0x3b3
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:52 +0x244
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:90 +0x367

Goroutine 40 (running) created at:
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:53 +0x16a
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.InitMetric()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/metric.go:101 +0xd04
  main.initTraceMetric.func2()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:377 +0x155
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:259 +0xd1
==================
==================
WARNING: DATA RACE
Read at 0x00c008dd2150 by goroutine 40:
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.(*Registry).ExportLog()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:49 +0x104
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).gatherAndWrite()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:85 +0x64
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:60 +0x164

Previous write at 0x00c008dd2150 by main goroutine:
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.(*Registry).Register()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:36 +0x164
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.Register()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:79 +0x213
  main.(*Config).createFileService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/config.go:213 +0x1cc
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:141 +0x11d
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:127 +0x3b3
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:52 +0x244
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:90 +0x367

Goroutine 40 (running) created at:
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:53 +0x16a
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.InitMetric()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/metric.go:101 +0xd04
  main.initTraceMetric.func2()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:377 +0x155
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:259 +0xd1
==================
==================
WARNING: DATA RACE
Read at 0x00c008d96200 by goroutine 40:
  github.com/matrixorigin/matrixone/pkg/perfcounter.(*CounterLogExporter).Export()
      /home/runner/work/matrixone/matrixone/head/pkg/perfcounter/counter_logexporter.go:38 +0x64
  github.com/matrixorigin/matrixone/pkg/util/metric/stats.(*Registry).ExportLog()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/stats/registry.go:49 +0x116
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).gatherAndWrite()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:85 +0x64
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:60 +0x164

Previous write at 0x00c008d96200 by main goroutine:
  github.com/matrixorigin/matrixone/pkg/perfcounter.NewCounterLogExporter()
      /home/runner/work/matrixone/matrixone/head/pkg/perfcounter/counter_logexporter.go:30 +0x677
  main.(*Config).createFileService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/config.go:211 +0x65e
  main.startService()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:141 +0x11d
  main.startCNServiceCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:127 +0x3b3
  main.startCluster()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/launch.go:52 +0x244
  main.main()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:90 +0x367

Goroutine 40 (running) created at:
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.(*StatsLogWriter).Start()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/stats_log_writer.go:53 +0x16a
  github.com/matrixorigin/matrixone/pkg/util/metric/mometric.InitMetric()
      /home/runner/work/matrixone/matrixone/head/pkg/util/metric/mometric/metric.go:101 +0xd04
  main.initTraceMetric.func2()
      /home/runner/work/matrixone/matrixone/head/cmd/mo-service/main.go:377 +0x155
  github.com/matrixorigin/matrixone/pkg/common/stopper.(*Stopper).doRunCancelableTask.func1()
      /home/runner/work/matrixone/matrixone/head/pkg/common/stopper/stopper.go:259 +0xd1
==================
2023/06/07 01:20:36.060408 +0000 INFO cnservice/distributed_tae.go:76 CN node running on push model.
```

### Expected Behavior

no race

**Comment by volgariver6** [comment]
@arjunsk please help to fix this.

**Comment by arjunsk** [comment]
This issue should be fixed in #9904.

**Comment by arjunsk** [comment]
Hi @sukki37 , This issue has been resolved in #9904 . Kindly close this issue.
